lib/domain/models/article.dart

```diff
@@ -0,0 +1,21 @@
+class Article {
+  final String title;
+  final String description;
+  final String author;
+  final String url;
+  final String urlToImage;
+  final DateTime publishedAt;
+
+  Article({required this.title, required this.description, required this.author, required this.url, required this.urlToImage, required this.publishedAt});
+
+  factory Article.fromJson(Map<String, dynamic> json) {
+    return Article(
+      title: json['title'],
+      description: json['description'],
+      author: json['author'] ?? 'Unknown',
+      url: json['url'],
+      urlToImage: json['urlToImage'],
+      publishedAt:  DateTime.parse(json['publishedAt']),
+    );
```

Comment:
try to secure it against the api changes, imagine if they changed the key `title` to `headline` so the app will show error and not represent any data, and imagine if one item only has `null` title, let's try to enhance this to handle those cases

Reply:
do you mean that i should check if the key exists and that the value is not null before setting them?